syft/pkg/cataloger/python/parse_requirements.go

```diff
@@ -22,37 +22,34 @@ func parseRequirementsTxt(_ string, reader io.Reader) ([]*pkg.Package, []artifac
 	scanner := bufio.NewScanner(reader)
 	for scanner.Scan() {
 		line := scanner.Text()
+		line = trimRequirementsTxtLine(line)
 
-		line = strings.TrimRight(line, "\n")
-
-		switch {
-		case strings.HasPrefix(line, "#"):
-			// commented out line, skip
+		if line == "" {
+			// nothing to parse on this line
 			continue
-		case strings.HasPrefix(line, "-e"):
+		}
+
+		if strings.HasPrefix(line, "-e") {
 			// editable packages aren't parsed (yet)
 			continue
-		case len(strings.Split(line, "==")) < 2:
-			// a package without a version, or a range (unpinned) which
-			// does not tell us exactly what will be installed
-			// XXX only needed if we want to log this, otherwise the next case catches it
-			continue
-		case len(strings.Split(line, "==")) == 2:
-			// remove comments if present
-			uncommented := removeTrailingComment(line)
-			// parse a new requirement
-			parts := strings.Split(uncommented, "==")
-			name := strings.TrimSpace(parts[0])
-			version := strings.TrimSpace(parts[1])
-			packages = append(packages, &pkg.Package{
-				Name:     name,
-				Version:  version,
-				Language: pkg.Python,
-				Type:     pkg.PythonPkg,
-			})
-		default:
+		}
+
+		if !strings.Contains(line, "==") {
+			// a package without a version, or a range (unpinned) which does not tell us
+			// exactly what will be installed.
 			continue
 		}
+
+		// parse a new requirement
+		parts := strings.Split(line, "==")
+		name := strings.TrimSpace(parts[0])
+		version := strings.TrimSpace(parts[1])
```

Comment:
would there be a case that the line is `something something ==` and still panics? the old check was `len(strings.Split(line, "==")) == 2`

Reply:
If I'm understanding correctly, the two implementations are logically equivalent. If you split the string `"foo =="` on `==`, you'll get a slice with two elements: `["foo ", ""]`.

So I think the answer to "can it still panic" is "no"... but let me know if I'm missing something!

Reply:
Right, then LGTM 👍